h2/src/main/org/h2/value/DataType.java

```diff
@@ -1383,8 +1345,47 @@ private static Value convertToValue1(SessionInterface session, Object x,
         } else if (clazz == Duration.class) {
             return JSR310Utils.durationToValue(x);
         } else {
-            return ValueJavaObject.getNoCopy(x, null, session.getDataHandler());
+            return convertToValueLobOrObject(session, x);
+        }
+    }
+
+    private static Value convertToValueLobOrObject(SessionInterface session, Object x) {
```

Comment:
the method name here "convertToValueLobOrObject" implies that we are converting to either a ValueLob or an Object, but I don't think that is what you mean.

Perhaps "convertObjectToValueLob"

Reply:
It returns either a `ValueLob` instance or a `ValueJavaObject` instance, but `convertToValueLobOrValueJavaObject` is too long from my point of view.